backend/keepalived/integration_test.go

```diff
@@ -55,23 +60,36 @@ func TestKeepaliveMonitor(t *testing.T) {
 		assert.FailNow(t, err.Error())
 	}
 
-	mFac := monitor.EtcdFactory(client, "TestKeepaliveMonitor")
-
-	k, err := New(Config{Store: store, Bus: bus, MonitorFactory: mFac})
-	require.NoError(t, err)
-
-	if err := k.Start(); err != nil {
-		assert.FailNow(t, err.Error())
+	if err := store.UpdateEntity(ctx, entity); err != nil {
+		t.Fatal(err)
 	}
 
-	entity := corev2.FixtureEntity("entity1")
-
 	keepalive := &corev2.Event{
-		Check:     &corev2.Check{Timeout: 1},
+		Check: &corev2.Check{
+			ObjectMeta: corev2.ObjectMeta{
+				Name:      "keepalive",
+				Namespace: "default",
+			},
+			Interval: 1,
+			Timeout:  5,
+		},
 		Entity:    entity,
 		Timestamp: time.Now().Unix(),
 	}
 
+	if err := store.UpdateEvent(ctx, keepalive); err != nil {
+		t.Fatal(err)
+	}
+
+	factory := liveness.EtcdFactory(context.Background(), client)
```

Comment:
Can we use the same context that the store uses on L52, or is there another reason we have independent contexts?

Reply:
That context carries extra information, and since it's being created by a function call in another package, I'd prefer to keep them separate.

Reply:
That makes sense, was just curious!